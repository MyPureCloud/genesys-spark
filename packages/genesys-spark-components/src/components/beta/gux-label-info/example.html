<h1>gux-label-info-beta</h1>

<h2>Default</h2>
<gux-label-info-beta>
  <span slot="content">This is an information tooltip</span>
</gux-label-info-beta>

<h2>Question</h2>
<gux-label-info-beta variant="question">
  <span slot="content">This is an information tooltip</span>
</gux-label-info-beta>

<h2>Info</h2>
<gux-label-info-beta variant="info">
  <span slot="content">This is an information tooltip</span>
</gux-label-info-beta>
