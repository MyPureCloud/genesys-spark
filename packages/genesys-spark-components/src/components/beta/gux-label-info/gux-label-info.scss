.gux-label-info {
  width: fit-content;
  padding: 0;
  line-height: 0;
  background-color: transparent;
  border: none;
  border-radius: var(--gse-ui-formControl-label-tooltipTrigger-borderRadius);

  gux-tooltip {
    padding-inline-start: 0;
  }

  & gux-icon {
    color: var(--gse-ui-formControl-label-tooltipTrigger-optionalColor);
    vertical-align: bottom;
  }
}
