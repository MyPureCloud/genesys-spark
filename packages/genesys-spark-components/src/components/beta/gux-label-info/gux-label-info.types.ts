export type GuxLabelInfoVariant = 'info' | 'question';
