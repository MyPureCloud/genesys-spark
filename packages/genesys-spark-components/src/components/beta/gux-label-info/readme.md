# gux-label-info-beta



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute   | Description | Type                                                                                                                                                                 | Default   |
| ----------- | ----------- | ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------- |
| `placement` | `placement` |             | `"bottom" \| "bottom-end" \| "bottom-start" \| "left" \| "left-end" \| "left-start" \| "right" \| "right-end" \| "right-start" \| "top" \| "top-end" \| "top-start"` | `'right'` |
| `variant`   | `variant`   |             | `"info" \| "question"`                                                                                                                                               | `'info'`  |


## Methods

### `hideTooltip() => Promise<void>`



#### Returns

Type: `Promise<void>`



### `showTooltip() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Slots

| Slot        | Description                                        |
| ----------- | -------------------------------------------------- |
| `"content"` | Required slot for tooltip and screenreader content |


## Dependencies

### Depends on

- [gux-screen-reader-beta](../gux-screen-reader)
- [gux-icon](../../stable/gux-icon)
- [gux-tooltip-beta](../gux-tooltip-beta)

### Graph
```mermaid
graph TD;
  gux-label-info-beta --> gux-screen-reader-beta
  gux-label-info-beta --> gux-icon
  gux-label-info-beta --> gux-tooltip-beta
  gux-tooltip-beta --> gux-tooltip-base-beta
  style gux-label-info-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
