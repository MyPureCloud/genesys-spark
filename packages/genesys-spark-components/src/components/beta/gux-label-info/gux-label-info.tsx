import { Component, Element, h, JSX, Prop, Method } from '@stencil/core';

import { Placement } from '@floating-ui/dom';

import { trackComponent } from '@utils/tracking/usage';
import { GuxLabelInfoVariant } from './gux-label-info.types';
import { getSlotTextContent } from '../../../utils/dom/get-slot-text-content';

/**
 * @slot content - Required slot for tooltip and screenreader content
 */

@Component({
  styleUrl: 'gux-label-info.scss',
  tag: 'gux-label-info-beta',
  shadow: true
})
export class GuxLabelInfo {
  private tooltipElement: HTMLGuxTooltipBetaElement;

  @Element()
  private root: HTMLElement;

  @Prop()
  variant: GuxLabelInfoVariant = 'info';

  @Prop({ mutable: true })
  placement: Placement = 'right';

  componentWillLoad(): void {
    trackComponent(this.root, { variant: this.variant });
  }

  private getVariantIcon(variant: GuxLabelInfoVariant): string {
    return variant === 'question'
      ? 'fa/circle-question-regular'
      : 'fa/circle-info-regular';
  }

  /*
   * Show tooltip
   */
  @Method()
  async showTooltip(): Promise<void> {
    return await this.tooltipElement.showTooltip();
  }

  /*
   * Hide tooltip
   */
  @Method()
  async hideTooltip(): Promise<void> {
    return await this.tooltipElement.hideTooltip();
  }

  render(): JSX.Element {
    return (
      <div class="gux-label-info">
        <gux-screen-reader-beta>
          <slot name="content"></slot>
        </gux-screen-reader-beta>
        <gux-icon
          icon-name={this.getVariantIcon(this.variant)}
          size="small"
          decorative
        ></gux-icon>
        <gux-tooltip-beta
          placement={this.placement}
          ref={el => (this.tooltipElement = el)}
        >
          <div slot="content">{getSlotTextContent(this.root, 'content')}</div>
        </gux-tooltip-beta>
      </div>
    ) as JSX.Element;
  }
}
