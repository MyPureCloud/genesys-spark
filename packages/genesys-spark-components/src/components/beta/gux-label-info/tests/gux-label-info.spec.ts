import { newSpecPage } from '@test/specTestUtils';
import { GuxLabelInfo } from '../gux-label-info';

const components = [GuxLabelInfo];
const language = 'en';

describe('gux-label-info-beta', () => {
  describe('#render', () => {
    [
      '<gux-label-info-beta></gux-label-info-beta>',
      '<gux-label-info-beta variant="question"><span slot="content">This is some tooltip text</span></gux-label-info-beta>',
      '<gux-label-info-beta variant="info"></gux-label-info-beta>',
      '<gux-label-info-beta><span slot="content">This is some tooltip text</span></gux-label-info-beta>',
      '<gux-label-info-beta variant="invalid-variant"><span slot="content">This is an information tooltip</span></gux-label-info-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
