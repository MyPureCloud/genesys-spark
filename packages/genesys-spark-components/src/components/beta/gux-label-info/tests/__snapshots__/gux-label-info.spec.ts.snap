// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-label-info-beta #render should render component as expected (1) 1`] = `
<gux-label-info-beta>
  <mock:shadow-root>
    <div class="gux-label-info">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/circle-info-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content"></div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
</gux-label-info-beta>
`;

exports[`gux-label-info-beta #render should render component as expected (2) 1`] = `
<gux-label-info-beta variant="question">
  <mock:shadow-root>
    <div class="gux-label-info">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/circle-question-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content">
          This is some tooltip text
        </div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
  <span slot="content">
    This is some tooltip text
  </span>
</gux-label-info-beta>
`;

exports[`gux-label-info-beta #render should render component as expected (3) 1`] = `
<gux-label-info-beta variant="info">
  <mock:shadow-root>
    <div class="gux-label-info">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/circle-info-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content"></div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
</gux-label-info-beta>
`;

exports[`gux-label-info-beta #render should render component as expected (4) 1`] = `
<gux-label-info-beta>
  <mock:shadow-root>
    <div class="gux-label-info">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/circle-info-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content">
          This is some tooltip text
        </div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
  <span slot="content">
    This is some tooltip text
  </span>
</gux-label-info-beta>
`;

exports[`gux-label-info-beta #render should render component as expected (5) 1`] = `
<gux-label-info-beta variant="invalid-variant">
  <mock:shadow-root>
    <div class="gux-label-info">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/circle-info-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content">
          This is an information tooltip
        </div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
  <span slot="content">
    This is an information tooltip
  </span>
</gux-label-info-beta>
`;
